master/internal/trials/postgres_trials.go

```diff
@@ -24,6 +25,48 @@ const (
 	batches = "batches"
 )
 
+type safeMetricToColumnMap struct {
+	m       map[string]string
+	reverse map[string]string
+}
+
+func newSafeMetricToColumnMap() safeMetricToColumnMap {
+	return safeMetricToColumnMap{
+		m:       make(map[string]string),
+		reverse: make(map[string]string),
+	}
+}
+
+func (s safeMetricToColumnMap) LookupOrAdd(key string) string {
+	if _, ok := s.m[key]; !ok {
+		for {
+			safeKey := generateMetricToColumn(key)
+			if _, ok := s.reverse[safeKey]; !ok {
+				s.m[key] = safeKey
+				s.reverse[safeKey] = key
+				break
+			}
+		}
+	}
+	return s.m[key]
+}
+
+func (s safeMetricToColumnMap) ReverseLookup(key string) string {
+	return s.reverse[key]
+}
+
+func generateMetricToColumn(metric string) string {
+	// Max length of a column name in postgres is 63 characters.
+	// Replace longer column names with a metric name prefix (for readability)
+	// and a uuid for randomness.
+	s := strings.ReplaceAll(metric, ".", "·")
```

Comment:
suggestion: this was for the sake of the pg driver and the resulting SQL query right? maybe mention that.

Reply:
added a comment.